controllers/etcdpeer_controller.go

```diff
@@ -63,18 +62,26 @@ func staticBootstrapInitialCluster(static etcdv1alpha1.StaticBootstrap) string {
 
 // advertiseURL builds the canonical URL of this peer from it's name and the
 // cluster name.
-func advertiseURL(etcdPeer etcdv1alpha1.EtcdPeer) *url.URL {
+func advertiseURL(etcdPeer etcdv1alpha1.EtcdPeer, port int32) *url.URL {
```

Comment:
nit: this function is named `advertiseURL`, but it's used to build things other than the advertiseURL since we added a `port` var. Perhaps rename to `buildPeerURL`?

It's also conceivable that etcd uses TLS for peer connections, but not for client, so the `scheme` parts here may need handling in future (or we may need specialised `buildClientURL(etcdPeer)` variants).

Can be left for now as it currently makes no functional difference, but it'd be good to have this named appropriately at least 😄

Reply:
> but it's used to build things other than the advertiseURL

Where? We only use it later on in the same file to populate the `ETCD_INITIAL_ADVERTISE_PEER_URLS` and `ETCD_ADVERTISE_CLIENT_URLS` environment variables. Both of which are "advertise urls" based on the language of etcd.

We only take in a port because both of those are different between the two (as they're advertising different things).